external/glfw/lib/android/android_init.c

```diff
@@ -837,16 +868,28 @@ int32_t glfwAndroidHandleInput(struct android_app* app, AInputEvent* event)
     int ret = 0;
     spinlock_lock(&g_EventLock);
 
-    if (g_NumAppInputEvents < MAX_APP_INPUT_EVENTS)
+    // We need to make sure we process all events in the queue, otherwise some gestures might end up
+    // in a wrong state and get stuck forever.
+    uint32_t all_event_count = countInputEvents(app, event);
+    if ((g_NumAppInputEvents + all_event_count) >= g_MaxAppInputEvents)
     {
-        // This will let the engine thread know (engine_main)
-        ret = addInputEvents(app, event, &g_AppInputEvents[g_NumAppInputEvents], &g_NumAppInputEvents, MAX_APP_INPUT_EVENTS);
-    }
-    else
-    {
-        LOGE("glfwAndroidHandleInput: max num app input events per frame reached");
+        uint32_t size_increase = APP_INPUT_EVENTS_SIZE_INCREASE_STEP * (uint32_t) ceil((float) all_event_count / (float) APP_INPUT_EVENTS_SIZE_INCREASE_STEP);
+        if (g_AppInputEvents == 0)
+        {
+            g_MaxAppInputEvents = size_increase;
+            g_AppInputEvents = malloc(sizeof(struct InputEvent) * g_MaxAppInputEvents);
+            memset(g_AppInputEvents, 0, sizeof(struct InputEvent) * g_MaxAppInputEvents);
+        }
+        else
+        {
+            g_MaxAppInputEvents += size_increase;
+            g_AppInputEvents = realloc(g_AppInputEvents, sizeof(struct InputEvent) * g_MaxAppInputEvents);
+        }
```

Comment:
No need for the check.
You can use `realloc`with a null pointer, so only the else statement is really needed.

Also, the first case zeroes out memory, and your second case doesn't.
I personally would skip the memset in favor of less code, as we're not likely to access out-of-bounds anyways, and if we did it would fail anyways.